ros/service_server.go

```diff
@@ -85,34 +87,38 @@ func (s *defaultServiceServer) start() {
 		s.node.waitGroup.Done()
 	}()
 
+	go s.chanLoop()
+
 	for {
-		//logger.Debug("defaultServiceServer.start loop");
-		s.listener.SetDeadline(time.Now().Add(1 * time.Millisecond))
-		if conn, err := s.listener.Accept(); err != nil {
-			opError, ok := err.(*net.OpError)
-			if !ok || !opError.Timeout() {
-				logger.Debugf("s.listner.Accept() failed")
-				return
-			}
-		} else {
-			logger.Debugf("Connected from %s", conn.RemoteAddr().String())
-			session := newRemoteClientSession(s, conn)
-			s.sessions.PushBack(session)
-			go session.start()
+		conn, err := s.listener.Accept()
+		if err != nil {
+			logger.Debugf("s.listner.Accept() failed")
```

Comment:
`s/listner/listener/`, please

are we making an assumption, that the only condition, under which "Accept" fails is when the underlying socket is closed?

Reply:
Good point. That's what we do for now. I guess it needs some more attention not only here but also in the similar place in ros/publisher.go. Unfortunately errors returned from listener.Accept() do not provide a simple way to differentiate connection closed from other cases. And listener doesn't export "IsClosed()" or similar interface, so the only way would be to have some sort of "closed" flag.